Spanner/src/SpannerClient.php

```diff
@@ -153,12 +162,32 @@ public function __construct(array $config = [])
             'returnInt64AsObject' => false,
             'projectIdRequired' => true,
             'hasEmulator' => (bool) $emulatorHost,
-            'emulatorHost' => $emulatorHost
+            'emulatorHost' => $emulatorHost,
+            'defaultQueryOptions' => null
         ];
 
         $this->connection = new Grpc($this->configureAuthentication($config));
         $this->returnInt64AsObject = $config['returnInt64AsObject'];
 
+        $this->defaultQueryOptions = $config['defaultQueryOptions'];
+
+        $envQueryOptions = new QueryOptions(
+            [
+                'optimizer_version' =>
+                    getenv('SPANNER_OPTIMIZER_VERSION') ?? ''
```

Comment:
This name is a bit vague. How about `SPANNER_QUERY_OPTIMIZER_VERSION`?

Reply:
We are trying to be consistent with the query hint which is just `OPTIMIZER_VERSION`. So unfortunately all the languages will need to use `SPANNER_OPTIMIZER_VERSION`.